### PR TITLE
fix(runner): `test.extend` doesn't work in hooks without test

### PR DESCRIPTION
**Pull request by Dunqing**
### Description

fix: #4030 

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: test/core/test/hooks-top-level.test.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { afterEach, beforeEach, expect, test as originalTest } from 'vitest'
+
+interface Fixture { foo: number }
+
+const test = originalTest.extend<Fixture>({
```
[the remark this answers is not in the document]
Comment (Dunqing) [reply test/core/test/hooks-top-level.test.ts]: > What happens in `beforeEah` if you call `extend` with the context that provides the same names? (Which is a viable option)
> 
> ```ts
> const test1 = test.extend({ foo: 1 })
> const test2 = test.extend({ foo: 2 })
> ```
> 

You can see the test cases, The context of `test1` and `test2` are unrelated.


> I think providing hooks on a test is a better alternative:
> 
> ```ts
> const test1 = test.extend({ foo: 1 })
> test1.beforeEach(({ foo }) => {
>   foo === 1
> })
> test1('contexted', ({ foo }) => {
>   expect(foo).toBe(1)
> })
> ```

In `vitest`, the `test1.beforeEach` doesn't support yet, the API supported in `playwright`

[the remark this answers is not in the document]
Comment (sheremet-va) [reply test/core/test/hooks-top-level.test.ts]: Ok then. API looks good to me, but tests are failing.

**Comment by Dunqing** [comment]
I think the test failure has nothing to do with this PR

**Comment by sheremet-va** [comment]
> I think the test failure has nothing to do with this PR

Only Windows and Mac tests are flaky, the Ubuntu one I see for the first time.

**Comment by Dunqing** [comment]
I figured out why, since concurrent causes an error caused by the fact that the test fetched by `getCurrentTest()` may not be correct. Now I get `fixtures` from context because the context is right in concurrent

**Pull request merged**